src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -1697,8 +1697,7 @@ void MergeTask::ExecuteAndFinalizeHorizontalPart::createMergedStream() const
             merge_parts_query_plan.getCurrentDataStream(),
             SizeLimits(), 0 /*limit_hint*/,
             global_ctx->deduplicate_by_columns,
-            false /*pre_distinct*/,
-            true /*optimize_distinct_in_order TODO: looks like it should be enabled*/);
+            false /*pre_distinct*/);
```

Comment:
In this case, we do not apply in order optimization anymore - which would lead to increased memory consumption during merges, at least

Reply:
Oh, that's a good catch!
Looks like we did not have a test for this :(